### ISSUE TITLE
Regression 1.1.54380. DB::Exception: Incorrect size of nested column in constructor of ColumnConst: 0, must be 1.

### ISSUE THREAD

**Issue by den-crane** [ClickHouse/ClickHouse#2314]
```
CREATE TABLE dim (dim_key Int64, dim_id String) ENGINE = MergeTree Order by (dim_key);

# cat dim.xml
<dictionaries>
    <dictionary>
        <name>dim</name>
        <source>
            <clickhouse><host>localhost</host><port>9000</port><user>default</user>
                <password></password><db>default</db><table>dim</table>
            </clickhouse>
        </source>
        <lifetime>0</lifetime>
        <layout> <complex_key_hashed/> </layout>
        <structure>
            <key><attribute><name>dim_key</name><type>Int64</type></attribute></key>
            <attribute><name>dim_id</name><type>String</type><null_value></null_value></attribute>
          </structure>
    </dictionary>
</dictionaries>


ClickHouse client version 1.1.54380.
Connecting to localhost:9000.
Connected to ClickHouse server version 1.1.54380.

:) SELECT dictGetString('dim', 'dim_id', tuple(toInt64(0)));
┌─dictGetString(\'dim\', \'dim_id\', tuple(toInt64(0)))─┐
│                                                       │
└───────────────────────────────────────────────────────┘
1 rows in set. Elapsed: 0.003 sec.

:) select dictGetString('dim', 'dim_id', x)  from (select tuple(toInt64(0)) as x);
Received exception from server (version 1.1.54380):
Code: 9. DB::Exception: Received from localhost:9000, ::1. DB::Exception: Incorrect size of nested column in constructor of ColumnConst: 0, must be 1..

0 rows in set. Elapsed: 0.059 sec.
```

And it worked in 1.1.54327

**Comment by alexey-milovidov** [comment]
I cannot reproduce it on master with your test case. Looks like this issue has been fixed.

**Comment by alexey-milovidov** [comment]
Confirmed.

**Comment by astudnev** [comment]
I met with the same bug on

SELECT
dictGetString('address_annotation', 'text', tuple(toUInt32(1), taker)) AS taker_annotation
FROM (
  SELECT
    '0x0000000000000000000000000000000000000000' taker
)

Code: 9. DB::Exception: Received from localhost:9000, ::1. DB::Exception: Incorrect size of nested column in constructor of ColumnConst: 0, must be 1.. 


Please fix, this is really stopper

**Comment by alexey-milovidov** [comment]
Is it still relevant in newer releases?

**Comment by den-crane** [comment]
Fixed in 18.16.0
